Core/Core/Quizzes/QuizDetails/ViewModel/QuizDetailsViewModel.swift

```diff
@@ -0,0 +1,126 @@
+    public var showSubmissions: Bool { course.first?.enrollments?.contains(where: { $0.isTeacher || $0.isTA }) == true }
+    public var assignmentSubmissionBreakdownViewModel: AssignmentSubmissionBreakdownViewModel?
+    public var quizSubmissionBreakdownViewModel: QuizSubmissionBreakdownViewModel?
+    public var assignmentDateSectionViewModel: AssignmentDateSectionViewModel?
+    public var quizDateSectionViewModel: QuizDateSectionViewModel?
+
+    public var quizTitle: String { quiz.first?.title ?? "" }
+    public var pointsPossibleText: String { quiz.first?.pointsPossibleText ?? "" }
+    public var published: Bool { quiz.first?.published ?? false }
+    public var quizDetailsHTML: String? { quiz.first?.details }
+    public var attributes: [QuizAttribute] {
+        guard let quiz = quiz.first else { return [] }
+        return QuizAttributes(quiz: quiz, assignment: assignment).attributes
+    }
+
+    private let quizID: String
+    private let courseID: String
+    private var assignment: Assignment?
+    private var refreshCompletion: (() -> Void)?
+    private lazy var course = env.subscribe(GetCourse(courseID: courseID)) { [weak self] in
+        self?.courseDidUpdate()
+    }
+
+    private lazy var quiz = env.subscribe(GetQuiz(courseID: courseID, quizID: quizID)) { [weak self] in
+        self?.didUpdate()
+    }
+
+    private lazy var assignments = env.subscribe(GetAssignmentsByGroup(courseID: courseID)) { [weak self] in
+        self?.didUpdate()
+    }
+
+    // MARK: - Public Interface -
+
+    public init(courseID: String, quizID: String) {
+        self.quizID = quizID
+        self.courseID = courseID
+    }
+
+    public func viewDidAppear() {
+        quiz.refresh(force: true)
+        assignments.refresh(force: true)
+        course.refresh()
+    }
+
+    public func editTapped(router: Router, viewController: WeakViewController) {
+        router.route(
+            to: "courses/\(courseID)/quizzes/\(quizID)/edit",
+            from: viewController,
+            options: .modal(.formSheet, isDismissable: false, embedInNav: true)
+        )
+    }
+
+    public func previewTapped(router: Router, viewController: WeakViewController) {
+        router.route(
+            to: "courses/\(courseID)/quizzes/\(quizID)/preview",
+            from: viewController,
+            options: .modal(.fullScreen, isDismissable: false, embedInNav: true)
+        )
+    }
+
+    // MARK: - Refreshable protocol
+
+    public func refresh(completion: @escaping () -> Void) {
+        refreshCompletion = completion
+        quiz.refresh(force: true)
+        assignments.refresh(force: true)
+    }
+
+    // MARK: - Private functions
+
+    private func courseDidUpdate() {
+        courseColor = course.first?.color
+    }
+
+    private func didUpdate() {
+        if quiz.requested, quiz.pending, assignments.requested, assignments.pending, assignments.hasNextPage { return }
+        finishRefresh()
```

Comment:
Might have no effect but I would call this at the end of the `didUpdate()` method when all local variables are up-to-date.